core/application/commands.go

```diff
@@ -122,3 +131,40 @@ func (a *app) DeleteMicroVM(ctx context.Context, uid string) error {
 
 	return nil
 }
+
+func (a *app) addInstanceData(vm *models.MicroVM, logger *logrus.Entry) error {
+	instanceData := &cloudinit.Metadata{}
+
+	meta, ok := vm.Spec.Metadata["meta-data"]
+	if ok {
```

Comment:
Could you drop the comma, ok mechanism here, and check for vm.Spec.Metadata not being ""?

You can decode `""` but it would fail the YAML decoding, which implies that `""` is not considered a valid value

Reply:
Changed the logic here based on your suggestion